hw/ip/otbn/doc/dv/index.md

```diff
@@ -805,6 +890,23 @@ The instruction-specific covergroup is `insn_bn_xid_cg` (shared with `BN.SID`).
 - Cross the three types of GPR for `grs1` with `grd_inc`
   Tracked in `enc_bnxid_cg` as `grs1_inc1_cross`.
 
+It is possible for BN.LID to trigger multiple errors in a single cycle.
+The possible errors are: underflow call stack (for `grs1`), underflow call stack (for `grd`), both increments set, invalid WDR index and bad data address.
+If we underflow the call stack for `grs1`, there's no architectural address, so that can't happen at the same time as a bad data address.
+Similarly, if we underflow the call stack for `grd`, there's no WDR index, so that can't cause an invalid WDR index.
+However, every other combination is possible.
```

Comment:
Setting both increments is considered an invalid instruction encoding so when seen no other errors will happen as execution just doesn't start.

To complicate matters an invalid WDR index also triggers an illegal instruction error but as it's a dynamic property of the instruction, other things could also occur (e.g. a call stack underflow).

Reply:
I understand that this is how the RTL works, but that's not explicit in the spec, which is (and should be!) the source document for this sort of DV test planning. What I've tried to do is list everything that looks like it might be possible if you read the ISA document. We don't allow things like underflowing and overflowing the call stack in a cycle since that's clearly impossible. But there are other designs that we could have built where the decoder merrily passes through a double increment and we catch it later. (The same goes for invalid CSRs, by the way)

Reply:
I see what you mean.

How do we plan to approach that from a coverage perspective? Whist you can generate code that in a design that say attempts the call stack pop in parallel with checking the increments are valid will produce both errors with our RTL it won't so you can't hit that as a coverpoint by RTL observation. As one of the errors is based upon dynamic state it can't be done via coverage over static properties of the instructions either. You'd need something to tell you an instruction would overflow the call stack and cross that with whether you have an illegal increment.

You can probe the RTL for this information but it could be a little fragile as it's not actually attempting the pop.

Reply:
Ahah :-) But, since the coverage collection is split out per instruction, it's not too difficult. You end up with logic like "The call stack was empty and grs1 was 1, so I would have underflowed". I've actually got a patch that does this and it's *nearly* hitting everything. I hope to push it up by the end of the day.